Equinor.Maintenance.API.EventEnhancer/Program.cs

```diff
@@ -1,18 +1,65 @@
+using System.Net.Mime;
+using Equinor.Maintenance.API.EventEnhancer.ConfigSections;
+using Equinor.Maintenance.API.EventEnhancer.Logging;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Identity.Web;
 using Serilog;
 
 var builder = WebApplication.CreateBuilder(args);
 
 // Add services to the container.
-builder.Host.UseSerilog((ctx, lc) => lc
-                                     .WriteTo.Console());
+builder.Services.AddTransient<HttpContextEnricher>();
+builder.Services.AddHttpContextAccessor();
+builder.Host.UseSerilog((_, services, lc) =>
+                        {
+                            lc
+                                .Enrich.FromLogContext()
+                                .Enrich.With(services.GetService<HttpContextEnricher>())
+                              .WriteTo.Console(outputTemplate: "[{Timestamp:HH:mm:ss} {Level:u3}] {Message:lj} by user {User}{NewLine}{Exception}");
+                        });
+
+builder.Services.AddOptions<AzureAd>()
+       .Bind(builder.Configuration.GetSection(nameof(AzureAd)))
+       .Validate(config => config.AllowedWebHookOrigins.Any(), "AzureConfig:AllowedWebHookOrigins must be populated")
+       .Validate(config => !string.IsNullOrWhiteSpace(config.ClientId), "AzureConfig:ClientId must be populated")
+       .Validate(config => !string.IsNullOrWhiteSpace(config.ClientSecret), "AzureConfig:ClientSecret must be populated")
+       .ValidateOnStart();
+
+
+builder.Services
+       .AddMicrosoftIdentityWebApiAuthentication(builder.Configuration, 
+                                                 subscribeToJwtBearerMiddlewareDiagnosticsEvents:builder.Environment.IsDevelopment());
 
 var app = builder.Build();
 
 // Configure the HTTP request pipeline.
 app.UseHttpsRedirection();
 
+app.UseAuthentication();
+app.UseAuthorization();
+
 app.MapGet("/", () => new OkObjectResult("Hello from Mini Api"))
 .WithName("HelloApi");
 
+app.MapMethods("/", new []{"OPTIONS"},

```

Comment:
Worth refactoring into it's own class in order to keep Program.cs light ?

Reply:
Yes, might implement the same CQRS pattern as in maintenance api, especially if the POST-code is large. Will refactor it at the same time in that case.

Reply:
Let's change the path to /maintenance-events

So that in APIM there'll be two methods in the OpenAPI.
OPTIONS /maintenance-events
POST /maintenance-events